src/App.js

```diff
@@ -1,50 +1,61 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import './App.scss';
 import {BrowserRouter as Router, Switch, Route} from 'react-router-dom';
-import Login from './pages/Login';
-import {GlobalProvider, GlobalContext} from './contexts/GlobalState';
+import {withAuthenticator, AmplifyContainer, AmplifyAuthenticator} from '@aws-amplify/ui-react';
+import {onAuthUIStateChange} from '@aws-amplify/ui-components';
+
+import {GlobalProvider} from './contexts/GlobalState';
 import Dashboard from './pages/Dashboard';
 import TemplatesView from './pages/TemplatesView';
 import TemplateWizard from './pages/TemplateWizard';
 import SubmitFile from './pages/SubmitFile';
 import FileView from './pages/FileView';
 import TemplateDesigner from './pages/TemplateDesigner';
 import NavMenu from './components/NavMenu';
+import {awsAuthInit} from './utils/auth';
 
 
+// Initialize Amplify
+awsAuthInit();
 /**
  * Main App builder
  * @return {Object} reactDOM
  */
-function App() {
-  const PrivateRoute = ({component: Component, ...attrs}) => (
-    <Route {...attrs} render={(props) => (
-      <GlobalContext.Consumer>
-        {(state) => (state.user.auth ? <Component route={props} {...props.match.params} /> : <Login {...props.match.params} /> )}
-      </GlobalContext.Consumer>
-    )} />
-  );
+const App = () => {
+  const [user, setUser] = useState({});
+  const [authState, setAuthState] = React.useState(null);
+
+  useEffect(() => {
+    return onAuthUIStateChange((nextAuthState, authData) => {
+      setAuthState(nextAuthState);
+      setUser(authData);
+    });
+  }, []);
+
   return (
-    <div className="App">
-      <GlobalProvider>
+    <GlobalProvider state={{user, authState}}>
+      <AmplifyContainer>
```

Comment:
AmplifyContainer replaces PrivateRoute/router logic, it requires login automatically